### PR TITLE
Fix EditStudentCommand bugs

### LINKED ISSUE #212: Alpha Bug: Edit Student
1. `Edited Student: Hi; Student ID: a2345678b; Phone: 99272758; Email: berniceyu@u.nus.edu; Tags: ` 
Should be edited student: **Student name: XX**

2. Include success and failure screenshots

3. Follow this format
`Acceptable values:

Index: Positive integer corresponding to the current displayed list from list-e.
(With at least one of the fields)
`

4. Matric number can be modified. But must check if got duplicate. Same for number, email

5. Prevent editing the StudentId if any loans/reservations exist.

### PR DESCRIPTION
**Pull request by junxiancheh**
closes #212 

Changes made:
1. If the current student is currently linked to any loans or record, the edit is blocked.

2. No edit is allowed if these fields already exist in addressbook: matrix number, phone number, email address.

3. More test cases

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/EditStudentCommandTest.java [hunk 1]
```diff
@@ -87,6 +92,58 @@ public void execute_invalidPersonIndexUnfilteredList_failure() {
         assertCommandFailure(editCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
     }
 
+    @Test
```
Comment (javnz1) [inline src/test/java/seedu/address/logic/commands/EditStudentCommandTest.java]: Have provided more test case to stress test the feature and its edge cases.
